src/main/java/com/telefonica/iot/cygnus/sinks/OrionMongoSink.java

```diff
@@ -122,13 +131,39 @@ void persist(Map<String, String> eventHeaders, NotifyContextRequest notification
                     backend.createCollection(dbName, collectionName);
                 } // if
 
-                LOGGER.info("[" + this.getName() + "] Persisting data at OrionMongoSink. Database: " + dbName
-                        + ", Collection: " + collectionName + ", Data: " + recvTimeTs / 1000 + "," + recvTime + ","
-                        + entityId + "," + entityType + "," + attrName + "," + entityType + "," + attrValue + ","
-                        + attrMetadata);
-                backend.insertContextDataRaw(dbName, collectionName, recvTimeTs / 1000, recvTime,
-                        entityId, entityType, attrName, attrType, attrValue, attrMetadata);
+                if (this.rowAttrPersistence) {
+                    LOGGER.info("[" + this.getName() + "] Persisting data at OrionMongoSink. Database: "
+                            + dbName + ", Collection: " + collectionName + ", Data: " + recvTimeTs.longValue() / 1000L
+                            + "," + recvTime + "," + entityId + "," + entityType + ","
+                            + attrName + "," + attrType + "," + attrValue + "," + attrMetadata);
+                    this.backend.insertContextDataRaw(
+                            dbName, collectionName, recvTimeTs.longValue() / 1000L, recvTime,
+                            entityId, entityType, attrName, attrType, attrValue, attrMetadata);
+                } else if (this.dataModel == DataModel.COLLECTIONPERATTRIBUTE) {
```

Comment:
If you see above, for me the COLLECTIONPERATTRIBUTE data model has no sense in the column-like persistence. Thus, this "else if" branch should probably removed.